apis/Google.Cloud.Logging.Log4Net/Google.Cloud.Logging.Log4Net/GoogleStackdriverAppender.cs

```diff
@@ -1,502 +1,508 @@
+                Timestamp = loggingEvent.TimeStamp.ToTimestamp(),
+                LogName = _logName,
+                Resource = _resource,
+                Labels = { labels },
+            };
+            if (sourceLocation != null)
+            {
+                logEntry.SourceLocation = sourceLocation;
+            }
+            return logEntry;
+        }
+
+        private void TryAddGitRevisionId(Dictionary<string, string> labels)
+        {
+            try
+            {
+                var gitId = SourceContext.AppSourceContext?.Git?.RevisionId;
+                if (!String.IsNullOrWhiteSpace(gitId))
+                {
+                    labels.Add(SourceContext.GitRevisionIdLogLabel, gitId);
+                }
+            }
+            catch (Exception ex) when (
+                ex is SecurityException 
+                || ex is InvalidProtocolBufferException
+                || ex is InvalidJsonException 
+                || ex is UnauthorizedAccessException)
+            {
+                // This is best-effort only, exceptions from reading/parsing the source_context.json are ignored.
+            }
+        }
+
+        private LogEntrySourceLocation BuildLogEntryLocation(LoggingEvent loggingEvent)
+        {
+            string file = null;
+            long? line = null;
+            string function = null;
+            if (loggingEvent.LocationInformation?.FileName != null)
+            {
+                file = loggingEvent.LocationInformation?.FileName;
+            }
+            if (long.TryParse(loggingEvent.LocationInformation?.LineNumber, out long lineNumber))
+            {
+                line = lineNumber;
+            }
+            string function0 = null;
+            string fullTypeName = loggingEvent.LocationInformation?.ClassName;
+            if (fullTypeName != null)
+            {
+                var type = _typeCache.GetOrAdd(fullTypeName, () =>
+                {
+                    try
+                    {
+#if NET45
+                            return AppDomain.CurrentDomain.GetAssemblies()
+                                .SelectMany(a => a.GetTypes())
+                                .FirstOrDefault(t => t.FullName == fullTypeName);
+#elif NETSTANDARD1_5
+                            // TODO: Support type lookup in netstandard
+                            return null;
+#else
+#error Unsupported platform.
+#endif
+                    }
+                    catch
+                    {
+                        // Ignore exceptions. This is best-effort only, and expected to fail in some situations.
+                        // Returning null caches the failure.
+                        return null;
+                    }
+                });
+                if (type != null)
+                {
+                    function0 = type.AssemblyQualifiedName;
+                }
+            }
+            string function1 = loggingEvent.LocationInformation?.MethodName;

```

Comment:
In the same way to `file`, `MethodName` also now contains "?" rather than being empty/null.
Please insert this code:
``` C#
if (function1 == "?")
{
    function1 = null;
}
```
following the assignment of `function1`.